File: examples/maven-spring-compile/src/main/java/community/flock/wirespec/example/maven/custom/app/user/UserAdapter.java
package community.flock.wirespec.example.maven.custom.app.user;

import java.util.List;

public interface UserAdapter {
    interface Module {
        UserAdapter userAdapter();
    }

    List<User> getAllUsers(String name);

    User getUserByName(final String name);

    User saveUser(final User user);

    User deleteUserByName(final String name);

    void uploadImage(final String name, byte[] bytes);
}

File: examples/gradle-ktor/src/main/kotlin/community/flock/wirespec/example/maven/custom/app/user/UserAdapter.kt
package community.flock.wirespec.example.maven.custom.app.user

interface HasUserAdapter {
    val userAdapter: UserAdapter
}

interface UserAdapter {
    fun getAllUsers(): List<User>

    fun getUserByName(name: String): User

    fun saveUser(user: User): User

    fun deleteUserByName(name: String): User
}
